src/extensions/web/Microsoft.DotNet.UpgradeAssistant.Extensions.Web/WebSdkCleanupAnalyzer.cs

```diff
@@ -37,7 +36,7 @@ public Task AnalyzeAsync(IProject project, IDependencyAnalysisState state, Cance
             // Check SDK directly (instead of using project.Components) since having the FrameworkReference counts as
             // having the AspNetCore component and this analyzer is specifically insterested in cases where both the SDK
             // and the framework reference are present.
-            if (!projectRoot.IsSdk || !projectRoot.Sdk.Equals(WebSdk, StringComparison.OrdinalIgnoreCase))
+            if (!projectRoot.Sdk.Contains(WebSdk, StringComparer.OrdinalIgnoreCase))
```

Comment:
Again, you should remove all the stringcomparer args here - you handle that in the collection